lib/absinthe/phase/operation/input.ex

```diff
@@ -0,0 +1,129 @@
+defmodule Absinthe.Phase.Operation.Input do
+  @moduledoc """
+  For a given operation name, provided a mapping of variable names to values:
+
+  - Add errors to any variable definition that is non-null but is not given a
+    value
+  - Set the `provided_value` field of all `Blueprint.Input.Argument.t` structs
+    that are using variables to the values provided for those variables (or
+    their defined defaults)
+  - Set the `provided_value` field of all `Blueprint.Input.Argument.t` structs
+    that are using literal values
+
+  ## Examples
+
+  Given a GraphQL document that looks like:
+
+  ```
+  query Item($id: ID!) {
+    item(id: $id, category: "Things") {
+      name
+    }
+  }
+  ```
+
+  And this phase configuration:
+
+  ```
+  run(blueprint, %{operation_name: "Item", variables: %{"id" => "1234"}})
+  ``
+
+  - The `item` field's `id` argument struct would be modified to have a
+    `provided_value` of `"1234"`.
+  - The `item` field's `category` argument struct would be modified to have a
+    `provided_value` of `"Things"`.
+
+  If, the following was used (note no `"id"` variable value is given):
+
+  ```
+  run(blueprint, %{operation_name: "Item", variables: %{}})
+  ``
+
+  - The variable definition for `id` would have an error record in its `errors`
+    field.
+  - The `item` field's `category` argument struct would be modified to have a
+    `provided_value` of `"Things"`.
+  """
+
+  alias Absinthe.{Blueprint, Phase}
+
+  @error_not_provided "value not provided"
+
+  @spec run(Blueprint.t, %{operation_name: nil | String.t, variables: %{String.t => any}}) :: {:ok, Blueprint.t}
+  def run(input, options) do
+    acc = Map.put(options, :active_operation, false)
```

Comment:
I think instead of doing this we should filter all selected operations from the blueprint operations list and then call prewalk on that. IE
```elixir
input.operations
|> Enum.filter(&(&1.name == options.operation_name)) # This easily allows batching
|> Blueprint.Mapper.prewalk(%{variables: options.variables}, &handle_node/2)
```